service/worker/dlq/workflow.go

```diff
@@ -123,22 +174,136 @@ func (c *workerComponent) deleteTasks(ctx context.Context, params DeleteParams)
 		},
 	}
 
-	_, err := c.client.DeleteDLQTasks(ctx, req)
+	_, err := c.historyClient.DeleteDLQTasks(ctx, req)
+	if err != nil {
+		return c.convertServerErr(err, "Failed to delete DLQ tasks")
+	}
+
+	return err
+}
+
+func (c *workerComponent) mergeTasks(ctx workflow.Context, params MergeParams) error {
+	params, err := parseParams(params)
 	if err != nil {
-		// Don't retry invalid-argument or not-found errors.
-		code := serviceerror.ToStatus(err).Code()
-		if code == codes.InvalidArgument || code == codes.NotFound {
-			return temporal.NewNonRetryableApplicationError(
-				"invalid argument deleting DLQ tasks", errorTypeInvalidRequest, err,
-			)
-		}
-		// Use the default retry policy for all other errors.
 		return err
 	}
+	ctx = workflow.WithActivityOptions(ctx, workflow.ActivityOptions{
```

Comment:
shall we add a retry policy for those activities?

Reply:
Done, please let me know if you think this retry policy is reasonable.